### PR TITLE
Add BE support for Doubles

### PR DESCRIPTION
**Pull request by directhex**
Missed from 41a6d3d6a5432b964b6f8439b6375ffad6f2e6cb

Author explicitly disclaims copyright ownership of change, as per https://lists.debian.org/debian-powerpc/2015/09/msg00002.html

### CODE REVIEW

File: mscorlib/system/globalization/charunicodeinfo.cs [hunk 1]
```diff
@@ -322,7 +338,7 @@ internal static bool IsWhiteSpace(char c)
             }
             return (((double*)s_pNumericValues)[pBytePtr[(ch & 0x000f)]]);
 #else
-            return (((double*)s_pNumericValues)[pBytePtr[(ch & 0x000f)]]);
+            return EndianSwap(((double*)s_pNumericValues)[pBytePtr[(ch & 0x000f)]]);
```
Comment (marek-safar) [inline mscorlib/system/globalization/charunicodeinfo.cs]: There is BitConverter.ToDouble to do this operation

Comment (directhex) [inline mscorlib/system/globalization/charunicodeinfo.cs]: Should the other calls to EndianSwap() from 41a6d3d6a5432b964b6f8439b6375ffad6f2e6cb be changed accordingly?

Comment (marek-safar) [inline mscorlib/system/globalization/charunicodeinfo.cs]: ideally, yes

Comment (nealef) [inline mscorlib/system/globalization/charunicodeinfo.cs]: Marek - ping?

Comment (marek-safar) [inline mscorlib/system/globalization/charunicodeinfo.cs]: @nealef for such case there is `Mono.Security.BitConverterLE.ToDouble`

Comment (nealef) [inline mscorlib/system/globalization/charunicodeinfo.cs]: Cool, I would like to make it a icall if it's not one already. For System z this operation reduces to a single instruction or two.

Comment (nealef) [inline mscorlib/system/globalization/charunicodeinfo.cs]: @marek-safar Also, note that there are several EndianSwap() methods already in that file as there was an earlier fix to this file relating to endian issues. It was in commit 41a6d3d6a5432b964b6f8439b6375ffad6f2e6cb. I assume these  invocations could also be changed to the Mono.Security.BitConverterLE.xxxx methods.

**Pull request merged**